torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -9,19 +11,39 @@ namespace fuser {
 namespace cuda {
 
 at::optional<CudaKernel*> CudaKernelCache::getKernelPtr(
-    const at::ArrayRef<c10::IValue> inputs) {
-  for (auto& iter : kernels_) {
-    if (iter.first->matchKernelSize(inputs)) {
-      return &(iter.second);
+    const at::ArrayRef<c10::IValue> inputs,
+    const std::vector<int64_t>& broadcasted_shape) {
+  for (auto& cuda_kernel : kernels_) {
```

Comment:
We iterate through all kernels and check if a given kernel could be used to run on the set of inputs.

TODO: we can cache the kernel selection for a set of input shapes, so we are not evaluating the flag every iteration. (maybe maintain a fixed-size table for kernel selection keys on input sizes)